addons/mrp/models/mrp_production.py

```diff
@@ -315,7 +315,7 @@ def _compute_date_deadline(self):
 
     def _set_date_deadline(self):
         for production in self:
-            (self.move_raw_ids | self.move_finished_ids).date_deadline = production.date_deadline
+            (production.move_raw_ids | production.move_finished_ids).date_deadline = production.date_deadline
```

Comment:
As there is propagation of date changes amongst linked moves, I believe it's more accurate to do this instead:
    production.move_finished_ids.date_deadline = production.date_deadline

The date on raw materials will be triggered after and recomputed preserving the manufacturing lead time.